File: api/news.py
# api/news.py
import os
import requests
import json
from datetime import datetime, timedelta

def handler(event, context):
    query_params = event.get("query") or {}
    query = query_params.get("query")
    if not query:
        return {"statusCode": 400, "body": "Query parameter missing"}

    API_KEY = os.environ.get("48d918617df64896b32e4687bc8b5aab")
    if not API_KEY:
        return {"statusCode": 500, "body": "API key not configured"}

    today = datetime.utcnow()
    two_days_ago = today - timedelta(days=2)

    params = {
        "q": query,
        "from": two_days_ago.strftime("%Y-%m-%d"),
        "to": today.strftime("%Y-%m-%d"),
        "sortBy": "publishedAt",
        "language": "en",
        "apiKey": API_KEY
    }

    try:
        res = requests.get("https://newsapi.org/v2/everything", params=params)
        data = res.json()
    except Exception as e:
        return {"statusCode": 500, "body": f"Error fetching news: {str(e)}"}

    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json", "Access-Control-Allow-Origin": "*"},
        "body": json.dumps(data)
    }

File: api/news.js
import fetch from "node-fetch";

export default async function handler(req, res) {
    const query = req.query.query;
    if (!query) return res.status(400).json({ error: "Query parameter missing" });

    const apiKey = process.env.NEWS_API_KEY;
    if (!apiKey) return res.status(500).json({ error: "API key not configured" });

    const today = new Date();
    const twoDaysAgo = new Date();
    twoDaysAgo.setDate(today.getDate() - 2);

    const url = `https://newsapi.org/v2/everything?q=${encodeURIComponent(query)}&from=${twoDaysAgo.toISOString().split("T")[0]}&to=${today.toISOString().split("T")[0]}&sortBy=publishedAt&language=en&apiKey=${apiKey}`;

    try {
        const response = await fetch(url);
        const data = await response.json();
        res.status(200).json(data);
    } catch (err) {
        res.status(500).json({ error: err.message });
    }
}
